src/annotator/guest.js

```diff
@@ -383,9 +382,12 @@ export default class Guest extends Delegator {
       return animationPromise(() => {
         const range = xpathRange.sniff(anchor.range);
         const normedRange = range.normalize(root);
-        const highlights = highlighter.highlightRange(normedRange);
-
-        $(highlights).data('annotation', anchor.annotation);
+        const highlights = /** @type {AnnotationHighlight[]} */ (highlightRange(
+          normedRange
+        ));
+        highlights.forEach(h => {
```

Comment:
What is the (higher-level) purpose here? I wonder if a comment could be handy?

Reply:
It's for linking highlights to their associated annotation so that, for example, we can look up which annotation card to highlight when the user hovers an annotation.